api_volontaria/apps/volunteer/views.py

```diff
@@ -157,3 +159,19 @@ class ParticipationViewSet(viewsets.ModelViewSet):
     }
     permission_classes = (DRYPermissions,)
     filter_backends = (ParticipationFilterBackend, DjangoFilterBackend)
+
+
+class TagViewSet(viewsets.ModelViewSet):
+
+    serializer_class = TagSerializer
+    queryset = Tag.objects.all()
+    filter_fields = '__all__'
+    permission_classes = (DRYPermissions,)
+
+    def get_permissions(self):
+        if self.action in ['list', 'retrieve']:
+            permission_classes = []
+        else:
+            permission_classes = [IsAdminUser]
+
+        return [permission() for permission in permission_classes]
```

Comment:
This logic should be put with [Dry-Rest-Permissions](https://github.com/fjnr-inc/dry-rest-permissions) format in order to protect the model directly

---

 - [Define permission logic on the model](https://github.com/fjnr-inc/dry-rest-permissions#define-permission-logic-on-the-model)
 - [Helpful decorators](https://github.com/fjnr-inc/dry-rest-permissions#helpful-decorators)

Reply:
I was following the code structure of the other ViewSet like CellViewSet in the same file.  Do you mean adding the functions like 'has_read_permission' and 'has_object_read_permission' in the Tag model ?

Reply:
Good point! I will make sure to correct these two bad usage of permission that you quote. In the same time you can take example on the [`position` app](https://github.com/Volontaria/API-Volontaria/tree/develop/api_volontaria/apps/position) that show a good usage of Dry-Rest-Permission